coderd/workspaces.go

```diff
@@ -1056,7 +1056,8 @@ func (api *API) putWorkspaceDormant(rw http.ResponseWriter, r *http.Request) {
 		if initiatorErr == nil && tmplErr == nil {
 			dormantTime := dbtime.Now().Add(time.Duration(tmpl.TimeTilDormant))
 			_, err = api.NotificationsEnqueuer.Enqueue(
```

Comment:
Is there an option to place `dbauthz.AsNotifier` inside `Enqueue()` or add some wrapper on `NotificationsEnqueuer`?

Reply:
Yes, that was an alternative option I was considering instead of forcing all of this onto callers. We can just have the notifier use the correct authz context internally and not have to worry about it. However, this means that _any_ actor can then enqueue _any_ kind of notification as long as it's plumbed through in code.

Reply:
I'll punt the decision for now and make it a potential follow-up since I'm not sure of the wider ramifications of this potential change yet.

Reply:
I think if we put `dbauthz.AsNotifier` inside `Enqueue` we're effectively working against the purpose of the RBAC system in my opinion.

Reply:
If only we can build a _magic_ wrapper that can assume the auth rule...